src/script/util/messageRenderer.ts

```diff
@@ -50,6 +50,24 @@ const originalFenceRule = markdownit.renderer.rules.fence!;
 
 markdownit.renderer.rules.heading_open = () => '<div class="md-heading">';
 markdownit.renderer.rules.heading_close = () => '</div>';
+const originalNormalizeLink = markdownit.normalizeLink!;
+
+const isValidUrl = (url: string): boolean => {
+  // only allow urls to https://, http:// and mailto:
+  return !!url.match(/^(https?:\/\/|mailto:)/i);
+};
+markdownit.validateLink = isValidUrl;
+markdownit.normalizeLink = (url: string): string => {
+  url = originalNormalizeLink(url);
+  if (isValidUrl(url)) {
+    return url;
+  }
+  // prepend "https://" if url does not contain a protocol
+  if (!url.match(/^[a-z]+:/i)) {
```

Comment:
is it forbidden to have numbers of uppercase letters in a protocol? (genuine question)

Reply:
good catch!

* uppercase is handled by making the regex case insensitive
* digits, punctuation and dashes (e.g. see https://www.iana.org/assignments/uri-schemes/uri-schemes.xhtml)

Without further fine-tuning, this currently means that 
`a6://something` would be converted to `https://a6://something` instead of being ignored